Source/CarthageKit/Project.swift

```diff
@@ -631,6 +648,17 @@ private func filesInDirectory(directoryURL: NSURL, typeIdentifier: String) -> Si
 		}
 }
 
+/// Sends the URL to each file found in the given directory that has the
+/// given file extension.
+///
+/// Note: If a proper UTI is defined for the file extension,
+///	      `filesInDirectory(_:typeIdentifier:)` is preferred.
+private func filesInDirectory(directoryURL: NSURL, #fileExtension: String) -> SignalProducer<NSURL, CarthageError> {
+	return NSFileManager.defaultManager().carthage_enumeratorAtURL(directoryURL, includingPropertiesForKeys: [], options: .SkipsHiddenFiles | .SkipsPackageDescendants, catchErrors: true)
```

Comment:
How about overloading `filesInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError>` or changing the existing one to `filesInDirectory(directoryURL: NSURL, typeIdentifier: String?) -> SignalProducer<NSURL, CarthageError>`, then filter the result from the caller?